arrow/formatter.py

```diff
@@ -116,7 +130,11 @@ def _format_token(self, dt, token):
         if token in ["ZZ", "Z"]:
             separator = ":" if token == "ZZ" else ""
             tz = dateutil_tz.tzutc() if dt.tzinfo is None else dt.tzinfo
-            total_minutes = int(util.total_seconds(tz.utcoffset(dt)) / 60)
+            utcoffset = tz.utcoffset(dt)
+            if utcoffset is None:
+                return None
```

Comment:
utcoffset should never be none in Arrow since we will always have aware datetime objects (never naive). According to the docs (https://docs.python.org/3/library/datetime.html#determining-if-an-object-is-aware-or-naive), `utc.offset` will never return None if a datetime object is aware. So I do not think we need to worry about utcoffset ever being None in our case. CC @systemcatch to hear his thoughts.

Reply:
I don't think we can end up with utcoffset being None here given the line above.

Reply:
`Arrow` passes `self._datetime` to this method and it can be naive as [discuss above](https://github.com/arrow-py/arrow/pull/883#discussion_r529185727).
And I also think this method should handle naive `datetime`, because `DateTimeFormatter.format` is public anybody can pass any `datetime` object.

Reply:
Yes `datetime` can be naive (None) but in that case we end up with the following.

```shell
(arrow) chris@ThinkPad:~/arrow$ python
Python 3.8.3 (default, Jul  7 2020, 18:57:36) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from dateutil import tz
>>> tutc=tz.tzutc()
>>> from datetime import datetime
>>> dt=datetime(2019, 5, 6, 7)
>>> utcoffset=tutc.utcoffset(dt)
>>> utcoffset
datetime.timedelta(0)
```

Reply:
@systemcatch 
I saw [Python datetime doc](https://docs.python.org/3/library/datetime.html#determining-if-an-object-is-aware-or-naive) and it saids `d.tzinfo.utcoffset(d) does not return None`. So I created a [test case](https://github.com/arrow-py/arrow/pull/883/files#diff-e0cd3f215f68d0608393c012e268cb2f896c1e88315c97fa77e114d400f5283bR134) for the scenario. please check it.

```
Python 3.8.6 (default, Sep 30 2020, 04:00:38) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from datetime import tzinfo, datetime
>>> class IncompleteTzinfo(tzinfo):
...   def utcoffset(self, dt):
...     return None
... 
>>> dt = datetime.utcnow()
>>> tz = IncompleteTzinfo()
>>> tz.utcoffset(dt)
>>> 

```

Reply:
Interesting, someone would need to make a subclass specifically to do this right?

I suppose this adds extra safety with no downside so let's leave it in.

Reply:
OK then I will remove this handling codes and just cast it to `timedelta`